### PR TITLE
Add tomte as a workflow in StatusDB

### PR DESCRIPTION
**Pull request by fevac**
### Added

- Add tomte as workflow to StatusDB

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Use git actions to deploy in stage
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b add_tomte_analysis -a
    ```

### How to test

- [x] Make sure workflow is added to statusDB

### Expected test outcome

- [x] Take a screenshot and attach or copy/paste the output.
image

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: alembic/versions/2024_03_14_b2fcb2ada306_add_tomte_to_analysis_options.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+def downgrade():
+    bind = op.get_bind()
+    session = sa.orm.Session(bind=bind)
+    for analysis in session.query(Analysis).filter(Analysis.workflow == new_workflow):
+        print(
+            f"Changing workflow for Analysis {Analysis.case.internal_id}, {Analysis.completed_at} to {downgrade_workflow_to}"
+        )
+        analysis.workflow = downgrade_workflow_to
```
Comment (henrikstranneheim) [inline alembic/versions/2024_03_14_b2fcb2ada306_add_tomte_to_analysis_options.py:81]: Nice! I do not think we should change any data. Just add tomte to the schema.

Comment (fevac) [reply alembic/versions/2024_03_14_b2fcb2ada306_add_tomte_to_analysis_options.py]: removed!

File: alembic/versions/2024_03_14_b2fcb2ada306_add_tomte_to_analysis_options.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def upgrade():
+    op.alter_column("case", "data_analysis", type_=new_analysis_enum)
+    op.alter_column("analysis", "workflow", type_=new_analysis_enum)
+
+
+def downgrade():
+    op.alter_column("case", "data_analysis", type_=old_analysis_enum)
+    op.alter_column("analysis", "workflow", type_=old_analysis_enum)
```
Comment (islean) [inline alembic/versions/2024_03_14_b2fcb2ada306_add_tomte_to_analysis_options.py:58]: The alter_column calls can crash if you don't specify existing_type, but I guess that we will see when this is deployed to hasta stage. Seems to be MYSQL specific https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.alter_column.

**Comment by fevac** [comment]
@islean @henrikstranneheim tests are passing. Can any of you approve?

**Review by islean — APPROVED** [review]
LGTM! 💪

**Comment by fevac** [comment]
- Deployed

image

image

**Comment by henrikstranneheim** [comment]
@fevac I think you should add tomte also to the schema of Application Limitations. I think it was forgotten in your last Alembic migration.

**Comment by fevac** [comment]
I'll open an issue and include it. Thanks

**Pull request merged**